tests/core/daemon/test_daemon.py

```diff
@@ -438,6 +476,123 @@ async def test_get_routes(mock_lonely_daemon):
+            False,
+            {
+                "success": True,
+                "wallet_addresses": {
+                    test_key_data.fingerprint: [
+                        {
+                            "address": "xch1zze67l3jgxuvyaxhjhu7326sezxxve7lgzvq0497ddggzhff7c9s2pdcwh",
+                            "hd_path": "m/12381/8444/2/0",
+                        },
+                    ],
+                },
+            },
+        ),
+        # specifying count and index should return the correct wallet addresses
+        (
+            {"fingerprints": [test_key_data.fingerprint], "count": 2, "index": 1},
+            False,
+            {
+                "success": True,
+                "wallet_addresses": {
+                    test_key_data.fingerprint: [
+                        {
+                            "address": "xch16jqcaguq27z8xvpu89j7eaqfzn6k89hdrrlm0rffku85n8n7m7sqqmmahh",
+                            "hd_path": "m/12381/8444/2/1",
+                        },
+                        {
+                            "address": "xch1955vj0gx5tqe7v5tceajn2p4z4pup8d4g2exs0cz4xjqses8ru6qu8zp3y",
+                            "hd_path": "m/12381/8444/2/2",
+                        },
+                    ]
+                },
+            },
+        ),
+        # specifying non_observer_derivation=True should return addresses for hardened derivations
+        (
+            {"fingerprints": [test_key_data.fingerprint], "non_observer_derivation": True},
+            False,
+            {
+                "success": True,
+                "wallet_addresses": {
+                    test_key_data.fingerprint: [
+                        {
+                            "address": "xch1k996a7h3agygjhqtrf0ycpa7wfd6k5ye2plkf54ukcmdj44gkqkq880l7n",
+                            "hd_path": "m/12381n/8444n/2n/0n",
+                        }
+                    ]
+                },
+            },
+        ),
+        # specifying a list of fingerprints with one invalid fingerprint will return an error
+        (
+            {"fingerprints": [999999]},
+            False,
+            {
+                "success": False,
+                "error": "key(s) not found for fingerprint(s) {999999}",
+            },
+        ),
+        (
+            {"fingerprints": [test_key_data.fingerprint]},
+            True,
+            {
+                "success": False,
+                "error": f"missing private key for key with fingerprint {test_key_data.fingerprint}",
+            },
+        ),
+    ],
+)
+@pytest.mark.asyncio
+async def test_get_wallet_addresses(
+    mock_daemon_with_config_and_keys, monkeypatch, rpc_request, pubkeys_only, expected_result
+):
+    daemon = mock_daemon_with_config_and_keys
+
+    original_get_keys = Keychain.get_keys
+
+    # monkeypatch Keychain.get_keys() to always call get_keys() with include_secrets=False
+    def mock_get_keys(self, include_secrets=False):
```

Comment:
I think this layer ended up not needed?

Reply:
Perhaps there's another way to write it, but this code is used and works.